framework/src/modules/README.md

```diff
@@ -1,32 +1,42 @@
 ## Modules
 
-Modules are individual building blocks for the lisk. The implementation of each module is up-to user but by default it should generate an object with this structure.
+Modules are individual building blocks for the lisk. The implementation of each module is up-to user but it must inherit from base module and implement the methods.
 
 ```js
 // Exported as main file to javascript package
-export default {
+export default class MyModule extends BaseModule {
+	/**
+	*
+ 	* @param {Object} options - An object of module options
+	*/
+	constructor(options) {
+	 super(options);
+	}
+
 	/**
 	 * A unique module name accessed through out the system.
 	 * If some module already registered with same alias, it will throw error
 	 */
-	alias: 'moduleAlias',
+	static get alias(){ return 'moduleAlias'; },
 
 	/**
 	 * Package detail referring the version and other details
 	 * Easiest way is to directly refer to package.json for all details
 	 */
 
-	info: {
-		author: '',
-		version: '',
-		name: '',
+	static get info(){
+		return {
+			author: '',
+			version: '',
+			name: '',
+			};
 	},
 
 	/**
 	 * Supported configurations for the module with default values
 	 */
 
-	defaults: {},
+	get defaults() { return {}; },
```

Comment:
I think if we only have `getter` it will throw an error if we do `* obj.defults = 'something'`
but it is very confusing to use `defaults` with getter because in babel/es rule you can export `default`. so,
```
export default class X {
  default: 'x',
}
```
and you can do
`import X from './above.js'`

Reply:
In this project unless we have babel or typescript we can not use `export default`. Event though its not confusing to if we have `const options = module.defaults`  and `export default`. Both have totally different reasons and syntax.

Reply:
@yatki getters are used if you don't want to a setter ;) 

Yes you can ask to freeze an object that will not be modified, but then it can not be modified by internal logic as well. So getters are a valid use case and in my experience its widely adopted language struct overall, for JS its newly announced language feature so will take some time for users to adopt.